ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java

```diff
@@ -3122,7 +3117,19 @@ Seems much cleaner if each stmt is identified as a particular HiveOperation (whi
     }
     return lockComponents;
   }
-  
+
+  private static LockType getLockTypeFromStorageHandler(WriteEntity output, Table t) {
+    final HiveStorageHandler storageHandler = Preconditions.checkNotNull(t.getStorageHandler(),
+        "Non-native tables must have an instance of storage handler.");
+    LockType lockType = storageHandler.getLockType(output);
+    if (null == LockType.findByValue(lockType.getValue())) {
```

Comment:
What's the benefit of doing `findByValue`? Isn't `if(null == lockType)` already sufficient?

Reply:
I haven't found any benefit :)
I just extracted these lines to a method to reuse this logic in case of insert overwrite.

Removed the findByValue part.